### PR TITLE
Trace2 tracing facility

### PR DESCRIPTION
**Pull request by jeffhostetler**
V7 addresses:
[] squashes in the fixup for windows process ancestry calculation.
[] adds optional socket_type parameter to the unix domain socket syntax.
[] allow unix domain socket syntax to try both stream and dgram if socket_type not specified.
[] add Perl prereq checks for JSON:PP in trace2 tests.
[] minor clang-format cleanup.

Thanks
Jeff

Cc: gitster@pobox.com
Cc: peff@peff.net
Cc: avarab@gmail.com
Cc: jeffhost@microsoft.com

### CODE REVIEW

**Comment by jeffhostetler** [comment]
@dscho Thanks for the info.  I'll give it another chance later after I do a little cleanup and re-push.

**Comment by dscho** [comment]
You're welcome. I submitted #109 to fix it.

**Comment by dscho** [comment]
From [the build log](https://dev.azure.com/gitgitgadget/git/_build/results?buildId=180&view=logs&jobId=fd490c07-0b22-5182-fac9-6d67fe1e939b&taskId=ce91d5d6-0c55-50f5-8ab9-6695c03ab102&lineStart=238&lineEnd=238&colStart=1&colEnd=107):

> 2019-01-18T01:25:12.3925336Z compat/win32/ancestry.c:3:10: fatal error: Psapi.h: No such file or directory
> 2019-01-18T01:25:12.3981024Z  #include <Psapi.h>
> 2019-01-18T01:25:12.4019672Z           ^~~~~~~~~

... whoops ;-)

This is a problem I introduced in my endeavor to minimize the subset of the Git for Windows SDK used for the continuous testing. I added that header to the sparse checkout file and am rebuilding the minimal SDK here: https://git-for-windows.visualstudio.com/git/_build/results?buildId=29013. Once that is done, I'll trigger another *CI for GitGitGadget's Git fork* run here.

**Comment by dscho** [comment]
@jeffhostetler I started a rebuild with the updated minimal SDK.

**Comment by dscho** [comment]
>  I started a rebuild

There you go. All green.

File: trace2/tr2_dst.c [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+}
+
+#ifndef NO_UNIX_SOCKETS
+#define PREFIX_AF_UNIX                 "af_unix:"
+#define PREFIX_AF_UNIX_STREAM          "af_unix:stream:"
+#define PREFIX_AF_UNIX_DGRAM           "af_unix:dgram:"
+
+static int tr2_dst_try_uds_connect(const char *path, int sock_type, int *out_fd)
+{
+	int e;
```
Comment (dscho) [inline trace2/tr2_dst.c]: Apparently this `e` variable [is unused](https://dev.azure.com/gitgitgadget/git/_build/results?buildId=2015&view=logs&jobId=8f20da19-31b7-5cef-4813-95b8788bd086&taskId=56027f08-fde3-50ad-0c9a-5ec7df432ed0&lineStart=180&lineEnd=184&colStart=1&colEnd=1) and should go...

**Pull request closed without merging**